src/Aspire.Hosting/ApplicationModel/WaitAnnotation.cs

```diff
@@ -48,3 +53,19 @@ public enum WaitType
     /// </summary>
     WaitForCompletion
 }
+
+/// <summary>
+/// Specifies the behavior of the wait.
+/// </summary>
+public enum WaitBehavior
+{
+    /// <summary>
+    /// If the dependency fails, ignore the failure and continue waiting.
+    /// </summary>
+    WaitOnDependencyFailure,
+
+    /// <summary>
+    /// If the dependency fails, stop waiting and fail the wait.
+    /// </summary>
+    StopOnDependencyFailure
```

Comment:
Terminal is more accurate than failure.

For example, someone stops the dependency doesn’t map well to failure.

Reply:
Can you suggest the rename?

Reply:
`StopOnDependencyTerminal`?

Or `StopOnDependencyUnavailable`?